cpmpy/solvers/pysat_rc2.py

```diff
@@ -0,0 +1,172 @@
+    pysat_vpool: a pysat.formula.IDPool for the variable mapping
+    pysat_solver: a pysat.solver.Solver() (default: glucose4)
+    """
+
+    @staticmethod
+    def supported():
+        # try to import the package
+        try:
+            import pysat
+            # there is actually a non-related 'pysat' package
+            # while we need the 'python-sat' package, some more checks:
+            from pysat.solvers import Solver
+            return True
+        except ImportError as e:
+            return False
+
+    @staticmethod
+    def solvernames():
+        """
+            Returns solvers supported by PySAT on your system
+        """
+        from pysat.solvers import SolverNames
+        names = []
+        for name, attr in vars(SolverNames).items():
+            if not name.startswith('__') and isinstance(attr, tuple):
+                if name not in attr:
+                    name = attr[-1]
+                names.append(name)
+        return names
+
+    def __init__(self, cpm_model=None, subsolver="glucose3"):
+        """
+        Constructor of the native solver object
+
+        Requires a CPMpy model as input, and will create the corresponding
+        PySAT clauses and solver object
+
+        Only supports satisfaction problems (no objective)
+
+        Arguments:
+        - cpm_model: Model(), a CPMpy Model(), optional
+        """
+        if not self.supported():
+            raise Exception("CPM_pysat: Install the python 'python-sat' package to use this solver interface (NOT the 'pysat' package!)")
+
+        from pysat.formula import WCNF
+        from pysat.examples.rc2 import RC2
+
+        assert subsolver in CPM_RC2.solvernames(), f"Wrong solver ({subsolver}) selected from available: ({CPM_RC2.solvernames()})"
+
+        self.pysat_solver = RC2(formula=WCNF(), solver=subsolver)
+        self.pysat_vpool = self.pysat_solver.pool
+
+        SolverInterface.__init__(self, name="rc2", cpm_model=cpm_model)
+
+    def solve(self, assumptions=None):
+        """
+            Call the PySAT solver
+
+            Arguments:
+            - time_limit:  maximum solve time in seconds (float, optional). Auto-interrups in case the
+                           runtime exceeds given time_limit.
+                           Warning: the time_limit is not very accurate at subsecond level
+            - assumptions: list of CPMpy Boolean variables that are assumed to be true.
+                           For use with s.get_core(): if the model is UNSAT, get_core() returns a small subset of assumption variables that are unsat together.
+                           Note: the PySAT interface is statefull, so you can incrementally call solve() with assumptions and it will reuse learned clauses
+        """
+        ## TODO: adding assumptions to the solving !
+        if assumptions is not None:
+            print("[Warning] Assumption variables interpreted as hard clauses!")
```

Comment:
I would remove this comment. Add it to the documentation. When repeated solving the terminal will be full of warnings...

Reply:
Also add an example on how to use the maxsat solver (for example the frietkot example, but add it to the examples list so you can refer to it from the documentation here).

Reply:
Added example